src/ngraph/runtime/hybrid/hybrid_backend.cpp

```diff
@@ -74,49 +44,149 @@ std::vector<T> read_vector(std::shared_ptr<ngraph::runtime::Tensor> tv)
-                                                                          const Shape& shape,
-                                                                          void* memory_pointer)
+shared_ptr<runtime::Tensor> runtime::hybrid::HybridBackend::create_tensor(
+    const element::Type& element_type, const Shape& shape, void* memory_pointer)
 {
-    return make_shared<runtime::HostTensor>(type, shape, memory_pointer, "external");
+    auto it = m_backend_list.begin();
+    return it->second->create_tensor(element_type, shape, memory_pointer);
 }
 
-bool runtime::hybrid::HYBRIDBackend::compile(shared_ptr<Function> function)
+bool runtime::hybrid::HybridBackend::compile(shared_ptr<Function> func)
 {
-    if (m_function_map.find(function) == m_function_map.end())
+    // auto it = m_backend_list.begin();
+    // return it->second->compile(func);
+    if (m_function_map.find(func) == m_function_map.end())
     {
+        vector<shared_ptr<runtime::Backend>> backend_list;
+        for (auto p : m_backend_list)
+        {
+            backend_list.push_back(p.second);
+        }
+
         // Clone function
         FunctionInstance instance;
-        instance.m_function = clone_function(*function);
+        instance.m_function = clone_function(*func);
 
+        // Run placement pass
         pass::Manager pass_manager;
+        pass_manager.register_pass<pass::AssignPlacement>(backend_list);
         pass_manager.run_passes(instance.m_function);
+
+        // Split function to sub_functions
+        tie(instance.m_sub_functions, instance.m_map_parameter_to_result) =
+            split_function_by_placement_size(instance.m_function);
+        m_function_map.insert({func, instance});
+
+        // Compile subfunctions in corresponding backends
+        for (shared_ptr<Function>& sub_function : instance.m_sub_functions)
+        {
+            size_t placement = get_colocated_function_placement_size(sub_function);
+            auto backend =
+                m_backend_list[(placement - 1)]; // (placement-1) as 0 is default placement
+            backend.second->compile(sub_function);
+        }
     }
+
     return true;
 }
 
-bool runtime::hybrid::HYBRIDBackend::call(shared_ptr<Function> function,
+bool runtime::hybrid::HybridBackend::call(shared_ptr<Function> func,
                                           const vector<shared_ptr<runtime::Tensor>>& outputs,
                                           const vector<shared_ptr<runtime::Tensor>>& inputs)
 {
-    validate_call(function, outputs, inputs);
+    // auto it = m_backend_list.begin();
+    // return it->second->call(func, outputs, inputs);
+    // Get FunctionInstance
+    bool rc = true;
+    auto it = m_function_map.find(func);
+    if (it == m_function_map.end())
+    {
+        compile(func);
+        it = m_function_map.find(func);
+    }
 
-    compile(function);
+   if (it == m_function_map.end())
+    {
+        throw runtime_error("Error constructing backend.");
```

Comment:
This needs to be a little more informative,

Reply:
This will never happen. The only way that the function map won't contain an instance is if the compile() function throws an exception in which case we won't even execute this code. I added a little more info to the message but there is no much to say since it really will never happen.